gcp/table_gcp_redis_instance.go

```diff
@@ -0,0 +1,360 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cloud.google.com/go/redis/apiv1/redispb"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+	"google.golang.org/api/iterator"
+)
+
+//// TABLE DEFINITION
+
+func tableGcpRedisInstance(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "gcp_redis_instance",
+		Description: "GCP Redis Instance",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"name", "location_id"}),
+			Hydrate:    getGcpRedisInstance,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listGcpRedisInstances,
```

Comment:
@madhushreeray30, Can we add the `location` column as an optional qualifier of the list config?
If someone wants to list available Redis Instances for a particular region, why should we make the API call for other regions?